mpp/include/mpp_util.inc

```diff
@@ -1574,9 +1583,13 @@ end function rarray_to_char
                    pnum_lines = 1
 
                    do
-                      read (UNIT=f_unit, FMT='(A)', IOSTAT=status) Content(pnum_lines)
+                      read (UNIT=f_unit, FMT='(A)', IOSTAT=status) str_tmp
```

Comment:
If this is done more than once, then `str_tmp` should be allocatable and allocated/deallocated every time.  If `len_trim(str_tmp) == LENGTH` is true the first time, it will be true every time because you never cleared the buffer.

Reply:
I tested this with intel and gfortran and it seems to work.